### PR TITLE
{Network} `az network watcher flow-log/troubleshooting`: Migrate to atomic commands

### PR DESCRIPTION
**Pull request by ZengTaoxu**
**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
`az network watcher flow-log configure`
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -5745,158 +5745,145 @@ def create_nw_packet_capture(cmd, client, resource_group_name, capture_name,
...
-        config.flow_analytics_configuration = None
-
-    with cmd.update_context(config) as c:
-        c.set_param('enabled', enabled if enabled is not None else config.enabled)
-        c.set_param('storage_id', storage_account or config.storage_id)
+        if not config['flowAnalyticsConfiguration']['networkWatcherFlowAnalyticsConfiguration']['workspaceId']:
+            config['flowAnalyticsConfiguration'] = None
+    except KeyError:
+        config['flowAnalyticsConfiguration'] = None
+    if enabled is not None:
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:5764]: What's the type of `enabled`? Is it string or bool type?

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 2]
```diff
@@ -5745,158 +5745,145 @@ def create_nw_packet_capture(cmd, client, resource_group_name, capture_name,
...
-                    c.set_param('workspace_id', workspace.properties['customerId'])
-                    c.set_param('workspace_region', workspace.location)
-                    c.set_param('workspace_resource_id', traffic_analytics_workspace)
-                    c.set_param('traffic_analytics_interval', traffic_analytics_interval)
-
-    return client.begin_set_flow_log_configuration(watcher_rg, watcher_name, config)
+                    nsg_flow_logging['flow_analytics_configuration']['workspace_id'] = workspace.properties['customerId']
+                    nsg_flow_logging['flow_analytics_configuration']['workspace_region'] = workspace.location
+                    nsg_flow_logging['flow_analytics_configuration']['workspace_resource_id'] = traffic_analytics_workspace
+                    nsg_flow_logging['flow_analytics_configuration']['traffic_analytics_interval'] = traffic_analytics_interval
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:5807]: It seems the property names of `config['flowAnalyticsConfiguration']['networkWatcherFlowAnalyticsConfiguration']` is camel case, however the last 4 lines use snack case name

File: src/azure-cli/azure/cli/command_modules/network/commands.py [hunk 3]
```diff
@@ -607,35 +595,24 @@ def load_command_table(self, _):
         g.command('stop', 'begin_stop')
         g.command('list', 'list')
 
-    with self.command_group('network watcher flow-log', client_factory=cf_network_watcher, min_api='2016-09-01') as g:
+    with self.command_group('network watcher flow-log') as g:
         g.custom_command('configure',
                          'set_nsg_flow_logging',
                          validator=process_nw_flow_log_set_namespace,
                          deprecate_info=self.deprecate(redirect='network watcher flow-log create', hide=False))
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/commands.py:602]: It has been announced for 3 years. We can remove this command.

File: src/azure-cli/azure/cli/command_modules/network/operations/watcher.py [hunk 4]
```diff
@@ -569,3 +574,370 @@ def pre_operations(self):
...
+        )
+        args_schema.storage_account._fmt = AAZResourceIdArgFormat(
+            template="/subscriptions/{subscription}/resourceGroups/{resource_group}/providers/Microsoft.Storage/storageAccounts/{}"
+        )
+
+        return args_schema
+
+    def pre_operations(self):
+        args = self.ctx.args
+        update_network_watcher_from_location(self.ctx,
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/operations/watcher.py:652]: Please use `get_network_watcher_from_location` instead,

File: src/azure-cli/azure/cli/command_modules/network/operations/watcher.py [hunk 5]
```diff
@@ -569,3 +574,370 @@ def pre_operations(self):
...
+        elif has_value(args.vnet) and not has_value(args.subnet):
+            args.target_resource_id = args.vnet
+        elif has_value(args.nic):
+            args.target_resource_id = args.nic
+        elif has_value(args.nsg):
+            args.target_resource_id = args.nsg
+
+        if has_value(args.retention):
+            if args.retention > 0:
+                args.retention_policy = {"days": args.retention, "enabled": (args.retention > 0)}
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/operations/watcher.py:675]:
```suggestion
                args.retention_policy = {"days": args.retention, "enabled": True}
```

File: src/azure-cli/azure/cli/command_modules/network/operations/watcher.py [hunk 6]
```diff
@@ -569,3 +574,370 @@ def pre_operations(self):
...
+            args.target_resource_id = args.subnet
+        elif has_value(args.vnet) and not has_value(args.subnet):
+            args.target_resource_id = args.vnet
+        elif has_value(args.nic):
+            args.target_resource_id = args.nic
+        elif has_value(args.nsg):
+            args.target_resource_id = args.nsg
+
+        if args.retention > 0:
+            args.retention_policy = {"days": args.retention, "enabled": (args.retention > 0)}
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/operations/watcher.py:788]:
```suggestion
            args.retention_policy = {"days": args.retention, "enabled": True}
```

File: src/azure-cli/azure/cli/command_modules/network/operations/watcher.py [hunk 7]
```diff
@@ -569,3 +574,370 @@ def pre_operations(self):
...
+                 "Only one Network Watcher can be existed per subscription and region.",
+            required=True,
+            fmt=AAZResourceLocationArgFormat(
+                resource_group_arg="resource_group",
+            ),
+        )
+        return args_schema
+
+    def pre_operations(self):
+        update_network_watcher_from_location(self.ctx,
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/operations/watcher.py:854]: Please use `get_network_watcher_from_location`

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**